src/com/amazon/ion/impl/IonReaderTextSystemX.java

```diff
@@ -480,20 +480,33 @@ private final void cast_cached_value(int new_type)
         final int count = _annotation_count;
         if (count == 0) return SymbolToken.EMPTY_ARRAY;
 
-        SymbolTable symbols = getSymbolTable();
+        resolveAnnotationSymbols();
 
         SymbolToken[] result = new SymbolToken[count];
-        for (int i = 0; i < count; i++)
-        {
-            SymbolToken sym = _annotations[i];
-            SymbolToken updated = _Private_Utils.localize(symbols, sym);
-            if (updated != sym) _annotations[i] = updated;
-            result[i] = updated;
-        }
+        System.arraycopy(_annotations, 0, result, 0, count);
 
         return result;
     }
 
+    public String[] getTypeAnnotations()
+    {
+        resolveAnnotationSymbols();
+        return _Private_Utils.toStrings(_annotations, _annotation_count);
+    }
+
+    /**
+     * Resolve annotations with the current symbol table.
+     */
+    private void resolveAnnotationSymbols() {
+        final int count = _annotation_count;
```

Comment:
interesting to note that we define a separate `count` variable, rather than just using `_annotation_count`.  is this avoiding situations in which another thread may alter the value of `_annotation_count`?

if so, should we be concerned that the value of `_annotation_count` might change between line 480 and 501?

i'm honestly not sure, but the final int caught my eye, and perhaps we should ensure we're honoring its intent

Reply:
TBH I'm not sure as well that's why I tried to mimic the existing code which uses a `count` variable in line 480. 

Taking a closer look it'd be safer to pass in the count to `resolveAnnotationSymbols` to really mimic the existing behavior

Reply:
Thanks for applying a safer/less-likely-to-break approach here